app/src/main/java/com/hunguet123/searchImage/ui/search/ResultSearchFragment.kt

```diff
@@ -0,0 +1,76 @@
+package com.hunguet123.searchImage.ui.search
+
+import android.app.AlertDialog
+import android.content.DialogInterface
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import com.example.searchimage.R
+import com.hunguet123.searchImage.data.model.Photo
+import com.hunguet123.searchImage.utils.Constant
+import kotlinx.android.synthetic.main.fragment_result_search.*
+import org.koin.androidx.viewmodel.ext.android.viewModel
+
+class ResultSearchFragment : Fragment(R.layout.fragment_result_search) {
+    private val viewModel : ResultSearchViewModel by viewModel()
+    private val photosAdapter = PhotosAdapter()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupViews()
+        val bundle = this.arguments
+        initData(bundle?.getString(Constant.SEARCH_BUNDLE_KEY))
+        listenEvents()
+        observeData()
+    }
+
+    private fun setupViews() {
+        recycleViewPhoto.adapter = photosAdapter
+    }
+
+    private fun listenEvents() {
+        photosAdapter.getAdapterPosition = { position ->
+            val alertDialog: AlertDialog? = activity?.let {
+                val builder = AlertDialog.Builder(it)
+                builder.apply {
+                    setMessage("Do you want add to favorite ?")
+                    setPositiveButton("YES",
+                        DialogInterface.OnClickListener { _, _ ->
+                            // User clicked OK button
+                            val url = viewModel.photosResponse.value?.hits?.get(position)?.URL.toString()
+                            println(url)
+                            val photoLocal = Photo(
+                                URL = url
+                            )
+                            viewModel.insertPhotoLocal(photoLocal)
+                        })
+                    setNegativeButton("NO",
+                        DialogInterface.OnClickListener { _, _ ->
+                            // User cancelled the dialog
+                        })
+                }
+                // Set other dialog properties
+
+                // Create the AlertDialog
+                builder.create()
+            }
+            alertDialog?.show()
+        }
+    }
+
+    private fun initData(q : String? ) {
+        viewModel.getImages(Constant.API_KEY, q, "photo")
+    }
+
+    private fun observeData() {
+        viewModel.photosResponse.observe(viewLifecycleOwner) { photosResponse ->
+            val listPhoto = photosResponse.hits ?: emptyList()
```

Comment:
phần xử lý response này đáng nhẽ ra xử lý nó từ Repository rồi 
Vì bản chất ViewModel cần là list Image chứ không phải toàn bộ response 
ở đây em chỉ nên observe list Hit 
viewModel.hits.....